src/service/userService.js

```diff
@@ -8,13 +8,77 @@ var backwardCompatibility = require("../util/backwardCompatibility");
 var passwordUtil = require("../util/password");
 
 module.exports = {
-  destroyAuthToken: function (user, callback) {
+  addOrUpdateFirebaseInstanceId (user, firebaseInstanceId, resultCallback) {
+  if (user && firebaseInstanceId) {
+    if (!user.firebaseInstanceIds){
+      user.firebaseInstanceIds = [];
+    }
+      async.detect(user.firebaseInstanceIds, (instanceIdTest, callback) => {
+      callback(null, instanceIdTest.instanceId === firebaseInstanceId);
+    }, (err, instanceIdFound) => {
+      if (err){
+        logger.log(err);
+      }
+      if(instanceIdFound){
+        logger.debug("Activating firebaseInstanceId: ", firebaseInstanceId, " for user: ", user.email);
+        instanceIdFound.lastUsedDate = Date.now();
+        instanceIdFound.active = true;
+      } else {
+        user.firebaseInstanceIds.push( { instanceId: firebaseInstanceId, active: true, createdDate: Date.now(), lastUsedDate: Date.now() } );
```

Comment:
What happened if async trigger this callback with `err`?
I guess we execute else branch with:
`instanceId: undefined`